### ISSUE TITLE
evaluate D

### ISSUE THREAD

**Issue by qingchuanhuajuan** [xinntao/ESRGAN#33]
I have some doubts about GAN.
D and G are alternately trained during the training.
How to evaluate that the D can make a correct judgment?Thank you！

**Comment by xinntao** [comment]
I think that D cannot make absolutely correct predictions as D is also improving itself during training. 
Typical losses of G and D can be found here: https://github.com/carpedm20/DCGAN-tensorflow (see Training details part)

I usually print the D outputs to see whether D is trained OK. If the outputs change suddenly or drastically, D usually behaviors badly.

**Comment by xinntao** [comment]
@JimmyChame 
At first of training, it will fluctuate a lot. With training, the outputs will be more stable.

**Comment by JimmyChame** [comment]
Thanks for your reply. I have another question if it doesn't interrupt.
I check the losses you mentioned in DCGAN-tensorflow, but I have a little doubt about them.
In the ideal case, the discriminator cannot distinguish between real and generated images at the end of training. The outputs of discriminator will be around 0.5, which means both of real and fake logits is 0. So I calculate the loss myself as following:

> `real = tf.ones_like(input)*0
fake = tf.ones_like(input)*0
real_loss = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(logits=real, labels=tf.ones_like(real)))
fake_loss = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(logits=fake, labels=tf.zeros_like(fake)))
d_loss = real_loss+fake_loss
g_loss = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(logits=fake, labels=tf.ones_like(fake)))`

Then I get the d_loss and g_loss values which are 1.3855876 and 0.6927938 separately. I found a large deviation between the value in DCGAN-tensorflow and the one I calculated. Am I wrong, or is this just an ideal case ?

**Comment by xinntao** [comment]
@JimmyChame It is just the ideal case.
